src/clr-angular/utils/focus/key-focus/key-focus-item.ts

```diff
@@ -11,7 +11,7 @@ import { isPlatformBrowser } from '@angular/common';
   selector: '[clrKeyFocusItem]',
 })
 export class ClrKeyFocusItem {
-  @HostBinding('attr.tabindex') tabIndex = -1;
+  @HostBinding('attr.tabindex') tabIndex: number;
```

Comment:
Does removing the default of `-1` make a difference?

Reply:
Yes. If users have some tabindexes specified on their elements, ours would override them by taking them out of the tab sequence.

Also, we needed to decouple tabindex management from focus management. They are two separate things that need to be managed differently. So let users manage their tabindex on their side, but we will take care of the focus management part on our side.

Reply:
1. I still have doubts if we do need to separate tabindex from focus. The main idea behind creating the KeyFocus directive was "separation of concern". I.e. separating the roving index pattern from the Tabs main business scenario. It made sense to be inside a directive, as focus in not handled in the angular view layer at all, and direct DOM manipulation was needed. Which is what directives are excelling at.
2. I agreed with Shijir that if this pattern is not accepted by the team, I am okay for it to go away. I just need to be sure that we understand what we are removing, why it was created this way, and that we refactor it for some obvious advantages.
3. I'd be happy to hear some comment from @coryrylan too if possible, who did the review of the directive initially and who created parts of the code in it, so he should have some good context about it.

That said - I don't mind the change. But if I have to do such a feature again, I'd still start the same way. I need to know why we prefer the "all control to the component" pattern.

Reply:
If users have some tabindexes specified on their elements, ours would override them by taking them out of the tab sequence. We cannot just assume users would implement roving index on their items all the time. Tabindex management would vary from component to component. By decoupling tabindex management from focus management, we are allowing more customizability for users and making the ClrKeyFocus more generic as it has now the better separation of concerns.

Reply:
Can you give me some practical examples of why users may need to set parallel tabindex management on the tabs? Not the content but on the tabs? Won't this be considered component hacking?
My example is that we now refuse to release modals without focus-grab, although users seem to ask for this once in a while. But we consider this to be a bad practice. Your point that the users _may_ want to have their own tab-index management of the tabs contradicts with our practices, as illustrated by the modals example.

Reply:
Why would it be considered component hacking? It's a separation of concerns; we are making it more generic by decoupling tabindex management. Why are we assuming a user would implement roving tabindex while using ClrKeyFocus? As a utility directive, it should do one thing and do it well, which in our case is to provide keyboard focus navigation. 

Again, users' tabindex management would vary from component to component. Users should have all the freedom to handle their own tabindex management. We shouldn't get in their way. In this specific case of tabs component, I noticed the exact problem where ClrKeyFocus provides roving tabindex, but conflicts with the tab component's manual activation.

As for the modal, that's a user component that is recommended to have a focus trap all the time. But ClrKeyFocus is a utility directive whose purpose is to provide keyboard focus navigation to other components. There is no authoring a11y guideline that says it has to come with its own roving tabindex management.

Reply:
This is an internal directive, and I'm of the mindset that generally a directive should be pretty limited in scope. The more work a particular entity does, the more complicated it is to reuse elsewhere. 

However, given it is internal and it worked fine in Angular 8, its not bad the way it is. Given the changes needed, I'm fine with this separation and think it might make it more reusable in other scenarios.

Reply:
We only have 2 scenarios - roving index and active descendant. I initially started with active descendant and even had the two in parallel for some time. They have nothing in common :) unless the goal they want to achieve.
That said - I can confirm that the last fix addresses the issue with skipping or blocking arrow key presses after shift-tab. I also confirmed with Chris that the shift-tab intermediate focus step is acceptable (though not required) - some authoritative aria examples have it, some don't. So it gets an approval from me too.
Thanks for the fix and the discussion.